examples/pytorch/rgcn/edge_classify.py

```diff
@@ -0,0 +1,490 @@
+                self.num_bases, activation=F.relu, self_loop=self.use_self_loop,
+                low_mem=self.low_mem, dropout=self.dropout, layer_norm = layer_norm))
+        # h2o
+        self.layers.append(RelGraphConv(
+            self.h_dim, self.out_dim, self.num_rels, "basis",
+            self.num_bases, activation=None,
+            self_loop=self.use_self_loop,
+            low_mem=self.low_mem, layer_norm = layer_norm))
+
+    def forward(self, blocks, feats, norm=None):
+        if blocks is None:
+            # full graph training
+            blocks = [self.g] * len(self.layers)
+        h = feats
+        for layer, block in zip(self.layers, blocks):
+            block = block.to(self.device)
+            h = layer(block, h, block.edata['etype'], block.edata['norm'])
+        return h
+
+
+class NeighborSampler:
+    """Neighbor sampler
+    Parameters
+    ----------
+    fanouts : list of int
+        Fanout of each hop starting from the seed nodes. If a fanout is None,
+        sample full neighbors.
+    """
+    def __init__(self, fanouts):
+
+        self.fanouts = fanouts
+
+    """Do neighbor sample
+    Parameters
+    ----------
+    seeds :
+        Seed nodes
+    Returns
+    -------
+    tensor
+        Seed nodes, also known as target nodes
+    blocks
+        Sampled subgraphs
+    """
+
+    def sample_blocks(self, g, seeds, *args, **kwargs):
+        blocks = []
+        cur = seeds
+        for fanout in self.fanouts:
+            if fanout is None or fanout == -1:
+                frontier = dgl.in_subgraph(g, cur)
+            else:
+                frontier = dgl.sampling.sample_neighbors(g, cur, fanout)
+            etypes = g.edata[dgl.ETYPE][frontier.edata[dgl.EID]]
+            block = dgl.to_block(frontier, cur)
+            block.srcdata[dgl.NTYPE] = g.ndata[dgl.NTYPE][block.srcdata[dgl.NID]]
+            block.srcdata['type_id'] = g.ndata[dgl.NID][block.srcdata[dgl.NID]]
+            block.edata['etype'] = etypes
+            cur = block.srcdata[dgl.NID]
+            blocks.insert(0, block)
+
+        return blocks
+
+
+# edge score prediction
+class ScorePredictor(nn.Module):
+    def __init__(self, num_classes, in_features, edge_features):
+        super().__init__()
+        self.edge_features = edge_features
+
+        edge_feature_dim = edge_features.shape[1]
+        h_dim = int(edge_feature_dim / 2)
+        print("edge_feature_dim: ", edge_feature_dim)
+        print("h_dim: ", h_dim)
+        dropout = 0.2
+
+        # add edge feature (edge features are converted to embeddings by a MLP
+        self.linear_add = th.nn.Sequential(
+            th.nn.Linear(edge_feature_dim, h_dim),
+            th.nn.ReLU(),
+            th.nn.Dropout(dropout)
+        )
+
+        self.linear = nn.Linear(2 * in_features + h_dim, num_classes)
+
+    def apply_edges(self, edges):
+        offer_id = edges.data['type_id'].tolist()
+        offer_features = self.edge_features[offer_id]
```

Comment:
Can we preprocess the edge_feature into torch tensor?
BTW, is this a common case to concatenate the edge feature? (Can we make this an optional choice?)

Reply:
Done. Added an option if the edge_feature is None, then only use node features to make predictions.